cpp/src/arrow/type.h

```diff
@@ -456,7 +456,7 @@ struct ARROW_EXPORT Date32Type : public DateType {
 
   Date32Type();
 
-  int bit_width() const override { return static_cast<int>(sizeof(c_type) * 4); }
+  int bit_width() const override { return static_cast<int>(sizeof(c_type) * 8); }
```

Comment:
I know there's some old school 7-bit platforms, but what's the relationship with the types in cstdint now with C++11 / POSIX standards?